pysrc/bytewax/connectors/kafka/serde.py

```diff
@@ -2,86 +2,57 @@
 import io
 import json
 import logging
-from abc import ABC, abstractmethod
-from typing import Dict, Generic, TypeVar
+from typing import Dict, Optional
 
-from confluent_kafka.schema_registry import record_subject_name_strategy
-from confluent_kafka.schema_registry.avro import AvroDeserializer, AvroSerializer
+from confluent_kafka.serialization import SerializationContext
 from fastavro import parse_schema, schemaless_reader, schemaless_writer
 from fastavro.types import AvroMessage
 
 from bytewax.connectors.kafka import MaybeStrBytes
 
 _logger = logging.getLogger(__name__)
 
-SerdeOut = TypeVar("SerdeOut")
-SerdeIn = TypeVar("SerdeIn")
 
+class PlainAvroSerializer:
+    """Bytewax serializer that uses fastavro's serializer.
 
-class SchemaSerializer(ABC, Generic[SerdeIn, SerdeOut]):
-    """A serializer for a specific schema."""
+    Beware that using plain avro serializers means you can't
+    deserialize with `confluent_kafka` python library, as that
+    requires some metadata prepended to each message, and plain
+    avro doesn't do that.
+    If you plan to deserialize messages with `confluent_kafka` python
+    library, please use the `ConfluentAvroSerializer` instead.
+    """
 
-    @abstractmethod
-    def ser(self, obj: SerdeIn) -> SerdeOut:
-        """Serialize an object."""
-        ...
+    def __init__(self, schema_str: str):
```

Comment:
Yes, I missed that, but `fastavro` does allow referencing schemas by name.
If you are reading schemas from a directory, `load_schema` automatically handles reading those.
You can also implement a light interface offered by `fastavro` to let it download schemas from a registry rather than looking at the directory (but you need to implement everything there).
Once you have all your referenced schemas parsed, you can pass them in a dictionary `named_schemas` to `parse_schema`.
We don't want to deal with retrieving the other schemas here, but it should be enough to proxy a `named_schemas` parameter to `parse_schema` for that to work. Fixing it